File: Euclid/Messages/Incoming/CLIENTIP.cs
using Euclid.Game;
using Euclid.Network.Session;
using Euclid.Network.Streams;

namespace Euclid.Messages.Incoming
{
    class CLIENTIP : IMessageEvent
    {
        #region Properties

        public bool AuthenticationRequired => false;

        #endregion

        public void Handle(Player player, Request request)
        {
            if (player.ConnectionMode == ConnectionMode.PUBLIC ||
                player.ConnectionMode == ConnectionMode.PRIVATE)
                player.Connection.InitialiseEncryption();
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/CLIENTIP.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.messages.IMessageEvent;
import euclid.network.session.ConnectionMode;
import euclid.network.streams.Request;

public class CLIENTIP implements IMessageEvent {

    @Override
    public boolean authenticationRequired() {
        return false;
    }

    @Override
    public void handle(Player player, Request request) {
        if (player.getConnectionMode() == ConnectionMode.PUBLIC ||
                player.getConnectionMode() == ConnectionMode.PRIVATE) {
            player.getConnection().initialiseEncryption();
        }
    }
}
